tests/random/prng_test.go

```diff
@@ -59,116 +118,312 @@ func TestRead(t *testing.T) {
-				var min, max, sum uint64
-				min = math.MaxUint64
-				for i, s := range samples {
-					if !s.IsUint64() {
-						t.Fatalf("sample[%d].IsUint64() got false; want true", i)
+					samples, err := ii.prng.Sample(nil, seed, uint16(tt.bitsPerSample), uint16(tt.numSamples))
+					if err != nil {
+						t.Fatalf("Sample(%x, 8 bits, 1000 samples) error %v", seed, err)
 					}
 
-					u := s.Uint64()
-					sum += u
-					if u < min {
-						min = u
-					}
-					if u > max {
-						max = u
+					var min, max, sum uint64
+					var onesCount int
+
+					numMax := (1 << tt.bitsPerSample) - 1
+					distMean := float64(numMax) / 2.
+					distVar := float64(numMax*numMax) / 12.
+
+					min = math.MaxUint64
+
+					var cusum, cusumMax int64
+
+					for i, s := range samples {
+						if !s.IsUint64() {
+							t.Fatalf("sample[%d].IsUint64() got false; want true", i)
+						}
+
+						u := s.Uint64()
+						sum += u
+						if u < min {
+							min = u
+						}
+						if u > max {
+							max = u
+						}
+
+						onesCount += bits.OnesCount64(u)
+
+						for j := uint64(0); j < tt.bitsPerSample; j++ {
+							z := 2*int64((u>>j)&1) - 1
+							cusum += z
+							abs := abs(cusum)
+							if abs > cusumMax {
+								cusumMax = abs
+							}
+						}
 					}
-				}
 
-				// Extremely unlikely that neither the min nor the max are
-				// returned given the high number of samples and the small
-				// sampling spaces.
-				if min != 0 {
-					t.Errorf("Min = %d; want 0", min)
-				}
-				if want := uint64(1<<tt.bitsPerSample) - 1; max != want {
-					t.Errorf("Max = %d; want %d", max, want)
-				}
+					t.Run("MinMax", func(t *testing.T) {
+						// Extremely unlikely that neither the min nor the max are
+						// returned given the high number of samples and the small
+						// sampling spaces.
+						if min != 0 {
+							t.Errorf("Min = %d; want 0", min)
+						}
+						if want := uint64(1<<tt.bitsPerSample) - 1; max != want {
+							t.Errorf("Max = %d; want %d", max, want)
+						}
+					})
 
-				expectedSum := (1<<(tt.bitsPerSample-1))*tt.numSamples - tt.numSamples/2
-				if dev := 1 - float64(expectedSum)/float64(sum); math.Abs(dev) > 0.01 {
-					t.Errorf("Sum = %s deviates by %.2f%% (>1%% abs) from expected sum %s", humanize.Comma(int64(sum)), dev*100, humanize.Comma(int64(expectedSum)))
-				}
-			})
+					t.Run("Sum", func(t *testing.T) {
 
-			t.Run("internal state", func(t *testing.T) {
-				// This test is primarily in place to confirm that the assembly
-				// implementation works as intended when converted to a
-				// high-level, more readable equivalent.
-				wantState := TestablePRNGState{
-					Seed:    new(big.Int).SetBytes(seed[:]),
-					Counter: big.NewInt(int64(tt.bitsPerSample*tt.numSamples/256 + 1)),
-					Remain:  big.NewInt(256 - int64(tt.bitsPerSample*tt.numSamples)%256),
-				}
+						expectedSum := distMean * float64(tt.numSamples)
+						// The variance of as sum of uniform random variables in
```

Comment:
(nit) `variance of a sum`

However the comment confused me because it made me think that the next variable would be the variance. Please add something like "we can use this to calculate a two-standard-deviation threshold for testing".